lib/spack/spack/store.py

```diff
@@ -70,11 +72,31 @@ def reindex(self):
 
 def _store():
     """Get the singleton store instance."""
-    root = spack.config.get('config:install_tree', default_root)
-    root = spack.util.path.canonicalize_path(root)
+    install_tree = spack.config.get('config:install_tree', {})
 
-    return Store(root,
-                 spack.config.get('config:install_path_scheme'),
+    if isinstance(install_tree, six.string_types):
+        tty.warn("Using deprecated format for configuring install_tree")
```

Comment:
I think we may want to add logic so that users can `spack config update config`. Since https://github.com/spack/spack/commit/193e8333fa23a2e9b44d44a80e153d9a27033860, you can add a function `update` to the `schema/config.py` which would do this automatically (and that commit added other logic to print useful error messages when there is a permissions issue).

(assuming you agree with that, a test for updating from old to new config would also be great)